src/set_theory/ordinal/arithmetic.lean

```diff
@@ -521,6 +521,9 @@ protected theorem add_sub_cancel_of_le {a b : ordinal} (h : b ≤ a) : b + (a -
   { exact (add_le_of_limit l).2 (λ c l, le_of_lt (lt_sub.1 l)) }
 end
 
+instance : has_exists_add_of_le ordinal :=
+⟨λ a b h, ⟨_, (ordinal.add_sub_cancel_of_le h).symm⟩⟩
```

Comment:
Is a stronger [canonically_ordered_add_monoid](https://leanprover-community.github.io/mathlib_docs/algebra/order/monoid.html#canonically_ordered_add_monoid) instance possible?

Reply:
Or even [canonically_linear_ordered_add_monoid](https://leanprover-community.github.io/mathlib_docs/set_theory/cardinal/basic.html#cardinal.canonically_linear_ordered_add_monoid)

Reply:
Neither of those requires addition to be commutative?

Reply:
They have the field `add_comm`.